server/src/main/java/org/elasticsearch/search/aggregations/support/MultiValuesSourceAggregationBuilder.java

```diff
@@ -225,7 +225,9 @@ protected boolean serializeTargetValueType() {
     public final XContentBuilder internalXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         if (fields != null) {
-            builder.field(CommonFields.FIELDS.getPreferredName(), fields);
+            for (Map.Entry<String, MultiValuesSourceFieldConfig> fieldEntry : fields.entrySet()) {
```

Comment:
++ I think the previous bit was just wrong and it wasn't noticed due to lack of to/from tests :)

Reply:
cool. I'll clean this up then, thanks